driver.go

```diff
@@ -123,8 +159,13 @@ func (l *lvmDriver) Create(req volume.Request) volume.Response {
 
 	v := &vol{Name: req.Name, MountPoint: mp}
 	if isSnapshot {
+		source := l.volumes[snap]
 		v.Type = "Snapshot"
 		v.Source = snap
+		v.KeyFile = source.KeyFile
```

Comment:
check if `keyFile` exists before updating `v.KeyFile = source.KeyFile`. We can use the utility function here.